src/publish_relay_message.rs

```diff
@@ -162,25 +156,72 @@ pub async fn subscribe_relay_topic(
 }
 
 #[instrument(skip(relay_client, metrics))]
-pub async fn extend_subscription_ttl(
+pub async fn batch_subscribe_relay_topics(
     relay_client: &Client,
-    topic: Topic,
+    topics: Vec<Topic>,
     metrics: Option<&Metrics>,
-) -> Result<(), Error<PublishError>> {
-    info!("extend_subscription_ttl");
-
-    // Extremely minor performance optimization with OnceLock to avoid allocating the same empty string everytime
-    static LOCK: OnceLock<Arc<str>> = OnceLock::new();
-    let message = LOCK.get_or_init(|| "".into()).clone();
-
-    let publish = Publish {
-        topic,
-        message,
-        tag: NOTIFY_NOOP_TAG,
-        ttl_secs: NOTIFY_NOOP_TTL.as_secs() as u32,
-        prompt: false,
+) -> Result<(), Error<SubscriptionError>> {
+    info!("batch_subscribe_relay_topic");
+    let start = Instant::now();
+
+    let call = || async {
+        let start = Instant::now();
+        let result = relay_client.batch_subscribe_blocking(topics.clone()).await;
+        if let Some(metrics) = metrics {
+            metrics.relay_batch_subscribe_request(start);
+        }
+        // TODO process each error individually
+        // TODO retry relay internal failures?
+        // https://github.com/WalletConnect/notify-server/issues/395
+        match result {
+            Ok(_) => Ok(()),
+            Err(e) => match e {
+                Error::Response(rpc::Error::Handler(
+                    SubscriptionError::SubscriberLimitExceeded,
+                )) => {
+                    // FIXME figure out how to handle this properly; being unable to subscribe means a broken state
```

Comment:
Consier creating an issue to address this and linking it here.

Reply:
This issue was root caused and no issue should currently exist because Notify Server always subscribes first. Confirmed with logs.

Removing this error handling logic now as the error should not happen.